browser/brave_browser_process_impl.cc

```diff
@@ -33,6 +34,17 @@ BraveBrowserProcessImpl::BraveBrowserProcessImpl(
       profile_creation_monitor_(new ProfileCreationMonitor) {
   g_browser_process = this;
   g_brave_browser_process = this;
+
+  brave_referrals_service_ = brave::BraveReferralsServiceFactory(local_state());
```

Comment:
I don't think we want to keep adding things to the BrowserProcessImpl like this. I don't think it's necessary, but also there is a larger problem here because this should be per-profile, not global. I think this should probably be a KeyedService like BraveRewards and should also go in components, not browser. See the `rewards` branch